packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js

```diff
@@ -1,29 +1,29 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 const util = require('@sap-cloud-sdk/util');
 const Constants = require('../utils/Constants');
-const { ValidationError } = require('../exceptions/validation-error');
-const { logger } = require('../logging/Logger');
-const ErrorStatuses = require('../utils/ErrorStatuses');
-const TenantConfigExtensionConstants = require('../utils/TenantConfigExtensionConstants');
+const { validateLength } = require('./ValidateLength');
+const { validatePattern } = require('./ValidatePattern');
+const destinationNameField = 'destinationName';
+const defaultDataSourceField = 'defaultDataSource';
+const defaultDataProviderCodeField = 'defaultDataProviderCode';
+const MAX_DESTINATION_NAME_LENGTH = 200;
+const MAX_VALUE_DATA_SOURCE_DATA_PROVIDER = 15;
 
-async function validatePrimaryCompositeKeysForTenantConfig(data) {
-  // validate the length of the dataProviderCode in the payload
+function validatePrimaryCompositeKeysForTenantConfig(data) {
   validateDefaultDataProviderCode(data.defaultDataProviderCode);
-  // validate the length of the dataSource in the payload
   validateDefaultDataSource(data.defaultDataSource);
   setIsConfigurationActiveIfNull(data);
   validateConnectionToMRM(data);
 }
 
 function validateConnectionToMRM(data) {
   if (!util.isNullish(data.connectToSAPMarketRatesManagement)) {
-    if (!Constants.DESTINATION_NAME_PATTERN.test(data.destinationName)) {
-      logger.error('The value for destinationName in the payload is invalid.');
-      throw new ValidationError(
-        TenantConfigExtensionConstants.INVALID_DESTINATION_NAME_FIELD_VALUE,
-        ErrorStatuses.BAD_REQUEST
-      );
-    }
+    validatePattern(
+      Constants.DESTINATION_NAME_PATTERN,
+      destinationNameField,
+      data.destinationName,
+      MAX_DESTINATION_NAME_LENGTH
```

Comment:
I had this point already on other PRs. I would really recommend to make the `validatePattern` generic in a sense that it tests a pattern for a given value and pattern. It uses then the field name and pattern info to make a nice error message. If your message is about the length than pass it in. But if the method stays like this it is not really usable in other places.

Reply:
All the errors resulting from the pattern check have details about the max length. So I'd like to keep the errors the same as that in the Java code. However, I have noted this and updated it in my issues and will relook at it in the next release.